### ISSUE TITLE
Syncing of chains between nodes in private mode has problems

### ISSUE THREAD

**Issue by jferas** [SushiChain/SushiChain#290]
## Checklist
- [X ] You've tried latest version of SushiChain
- [ X] You've confirmed there is not a same report

## Description
Nodes will properly sync with the target node given on the command line (via the -n option) and acquire their chain, but new blocks that are mined on the target don't seem to get sent from the target to the new node when running in private mode.

## Environment

- Using docker: no
- Crystal version: 0.29.0
- Where do you connect to? http://localhost:3000, 3001, and 3002 for the test

## Command Lines and results
(each sushid node and miner run from their own command shell windows and separate directories with their own `sushi_chain.db`)
```bash
./sushid -w testnet-wallet-1.json -d sushi_chain.db --testnet  --private

../sushid -w ../testnet-wallet-2.json -d sushi_chain.db --testnet  --private  -p 3001 -n http://0.0.0.0:3000

../sushid -w ../testnet-wallet-3.json -d sushi_chain.db --testnet  --private  -p 3002 -n http://0.0.0.0:3000

./sushim -w testnet-wallet-1.json --testnet -n http://127.0.0.1:3000 --process=2
```
The last 2 nodes start up and and sync with the first to get the full chain from it.. but...
newly mined blocks are put in the chain of the first node, but it (for some reason) doesn't see the connected nodes as predecessors or successors and doesn't send the new blocks to them.

**Comment by kingsleyh** [comment]
This is a low priority issue - which I think was a design choice. So we could just prevent any nodes from connecting to a node which is running in private mode - this would make it clear that this kind of configuration is not intended.

**Comment by kingsleyh** [comment]
This will be addressed by #296
